src/test/cypress/integration/quiz-exercise/QuizExercise.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { artemis } from '../../support/ArtemisTesting';
+import { generateUUID } from '../../support/utils';
+import { POST } from '../../support/constants';
+
+// Accounts
+const admin = artemis.users.getAdmin();
+
+// Requests
+const courseManagementRequest = artemis.requests.courseManagement;
+
+// PageObjects
+const navigationBar = artemis.pageobjects.navigationBar;
+const courseManagement = artemis.pageobjects.courseManagement;
+const quizCreation = artemis.pageobjects.quizExerciseCreation;
+
+// Common primitives
+let uid: string;
+let courseName: string;
+let courseShortName: string;
+let quizExerciseName: string;
+
+describe('Quiz Exercise Management', () => {
+    let course: any;
+
+    before('Set up course', () => {
+        uid = generateUUID();
+        courseName = 'Cypress course' + uid;
+        courseShortName = 'cypress' + uid;
+        cy.login(admin);
+        courseManagementRequest.createCourse(courseName, courseShortName).then((response) => {
+            course = response.body;
+        });
+    });
+
+    beforeEach('New UID', () => {
+        uid = generateUUID();
+        quizExerciseName = 'Cypress Quiz ' + uid;
+        cy.intercept(POST, '/api/quiz-exercises').as('createQuizExercise');
+    });
+
+    after('Delete Course', () => {
+        courseManagementRequest.deleteCourse(course.id);
+    });
+
+    describe('Quiz Exercise Creation', () => {
+        let quizExercise: any;
+
+        afterEach('Delete Quiz', () => {
+            courseManagementRequest.deleteQuizExercise(quizExercise.id);
+        });
+
+        it('Creates a Quiz with Multiple Choice', () => {
+            beginQuizCreation();
+            quizCreation.addMultipleChoiceQuestion('MC Quiz' + uid);
+            quizCreation.saveQuiz();
+            cy.wait('@createQuizExercise').then((quizResponse) => {
+                quizExercise = quizResponse?.response?.body;
+                cy.visit('/course-management/' + course.id + '/quiz-exercises/' + quizExercise.id + '/preview');
+                cy.contains('MC Quiz' + uid).should('be.visible');
+            });
```

Comment:
The `cy.wait()` part should be moved into the pageobject. The function then simply returns the chainable of the wait
```suggestion
            quizCreation.saveQuiz().then((quizResponse) => {
                quizExercise = quizResponse.response.body;
                cy.visit('/course-management/' + course.id + '/quiz-exercises/' + quizExercise.id + '/preview');
                cy.contains('MC Quiz' + uid).should('be.visible');
            });
```

Reply:
There is a <a href="https://github.com/ls1intum/Artemis/pull/3969">follow up</a> to this PR in which these will be addressed.